plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java

```diff
@@ -462,6 +465,18 @@ private void checkLanguageServerInitialized() {
     }
   }
 
+  private CompletableFuture<ServerCapabilities> getOrInitializeLanguageServer() {
```

Comment:
We should not do this. Whether we decide to initialize a server or not should be done by the server initializer. Don't cache the result, it's not worth it.